crates/sui-framework/sources/governance/sui_system.move

```diff
@@ -284,9 +288,14 @@ module sui::sui_system {
         let storage_fund = balance::value(&self.storage_fund);
         let total_stake = delegation_stake + validator_stake + storage_fund;
 
-        let delegator_reward_amount = delegation_stake * computation_charge / total_stake;
-        let delegator_reward = balance::split(&mut computation_reward, delegator_reward_amount);
+        let storage_fund_reward_amount = computation_charge * storage_fund / total_stake;
+        let storage_fund_reinvest_amount = storage_fund_reward_amount * self.parameters.storage_fund_reinvest_bps / 10000;
+        let storage_fund_reinvest = balance::split(&mut computation_reward, storage_fund_reinvest_amount);
         balance::join(&mut self.storage_fund, storage_reward);
+        balance::join(&mut self.storage_fund, storage_fund_reinvest);
+
+        let delegator_reward_amount = computation_charge * delegation_stake / total_stake;
```

Comment:
Not for the current PR, but a TODO for the future after our code is more stable (after our potential redesigin) is that we should write comprehensive tests for rewards distribution with different parameters. All these mathematical calculations are quite error-prone.

Reply:
Yeah totally agree.